internal/controller/pkg/revision/deployment.go

```diff
@@ -63,7 +63,6 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      provider.GetName(),
 					Namespace: namespace,
-					Labels:    map[string]string{"pkg.crossplane.io/revision": revision.GetName()},
```

Comment:
Similar/related to the other comment, we would lose this label if no controller config is provided.